brainframe_qt/ui/dialogs/task_configuration/core/zone.py

```diff
@@ -0,0 +1,154 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
```

Comment:
![image](https://user-images.githubusercontent.com/1740412/129815222-5d1635aa-cb2f-4784-a5d9-fd1afd399e6f.png)
I believe that previously, the Screen was always first in the list. I think if it's no big deal to add, it might be nice to keep that 'feature'.

Reply:
Umm weird... I didn't think I changed anything that would affect ordering here. I also haven't seen `Screen` anywhere but the top. I'll investigate

Reply:
So I think the only way this could happen is if `api.get_zones` returns `Screen` _not_ as the first Zone. Do you know why that would happen?

Reply:
I tested it, and it appears that api.get_zones() returns Screen first, but api.get_zones(stream_id=...) returns Screen last (or maybe everything just in inverse order?).

I only have two streams of sample size, but hey. 

Since this is more of a UI thing, we probably don't want to rely on the server to maintain this visual nicety. We could add this after the api.get_zones() call, to ensure the Screen zone is always first:

```
zones = list(map(Zone.from_api_zone, api_zones))

# These lines
screen_index = next(i for i, z in enumerate(zones) if z.name == FULL_FRAME_ZONE_NAME)
zones.insert(0, zones.pop(screen_index))
```

Reply:
Anyways, I approved the PR, but this would be a visual nicety to have. It makes sense to have the "always persistent" uneditable zone at the top of the list always.

Reply:
Well, yeah, it changed in the sense that now Screen is always at the bottom as opposed to always being at the top. 

I consider it a minor UI regression, because I think most brainframe users have gotten used to the screen being right at the top

Reply:
This should be resolved now